src/shortcut.rs

```diff
@@ -0,0 +1,52 @@
+#[allow(unused_imports)]
+use std::ffi::CString;
+use std::path::PathBuf;
+#[allow(unused_imports)]
+use std::process::Command;
+
+#[cfg(target_os = "windows")]
+#[link(name = "shell32")]
+extern "system" {
+    fn ShellExecuteA(
+        hwnd: isize,
+        lpOperation: *const i8,
+        lpFile: *const i8,
+        lpParameters: *const i8,
+        lpDirectory: *const i8,
+        nShowCmd: i32,
+    ) -> isize;
+}
```

Comment:
I suspect this works, but it is not a good way to do it. Why is command spawning not sufficient? If you do need to use such APIs, you should do it using https://crates.io/crates/windows that already exposes correct FFI and linking flag, no need to re-invent it. For example this function: https://microsoft.github.io/windows-docs-rs/doc/windows/Win32/UI/Shell/fn.ShellExecuteA.html

And while at it, probably should use `ShellExecuteW` to support unicode characters.

Reply:
I don't want to introduce a crate for a small function.

Reply:
I feel like this is an appropriate place to use a crate because you need this feature for 3 platforms and it involves unsafe and direct linking to system libraries. It is not *that* trivial actually. So I'd prefer to offload it to well maintained library rather than re-implementing here.

Reply:
ok I will switch to use open crate